apps/vast/vast-bridge.py

```diff
@@ -56,32 +68,75 @@ async def start(cmd, vast_endpoint, zmq_endpoint, snapshot):
     if not pub_endpoint or not sub_endpoint or not topic:
         logger.error("Unparsable subscription reply")
         exit(1)
-    logger.info(f"Subscription successfull")
+    logger.debug(f"Subscription successfull")
+    atexit.register(unsubscribe, zmq_endpoint, topic)
 
-    intel_task = asyncio.create_task(
-        receive_intel(cmd, vast_endpoint, pub_endpoint, topic)
+    intel_queue = asyncio.Queue()
+    sightings_queue = asyncio.Queue()
+    report_sightings_task = asyncio.create_task(
+        report_sightings(sub_endpoint, sightings_queue, transform_cmd, sink)
+    )
+    atexit.register(report_sightings_task.cancel)
+    receive_intel_task = asyncio.create_task(
+        receive_intel(pub_endpoint, topic, intel_queue)
     )
-    sighting_task = asyncio.create_task(
-        report_sightings(cmd, vast_endpoint, sub_endpoint)
+    atexit.register(receive_intel_task.cancel)
+    match_task = asyncio.create_task(
+        match_intel(cmd, vast_endpoint, intel_queue, sightings_queue, retro_match)
     )
+    atexit.register(match_task.cancel)
+    if not retro_match:
+        live_match_vast_task = asyncio.create_task(
+            live_match_vast(cmd, vast_endpoint, sightings_queue)
+        )
+        atexit.register(live_match_vast_task.cancel)
+        await asyncio.gather(
+            receive_intel_task, report_sightings_task, match_task, live_match_vast_task
+        )
+    else:
+        await asyncio.gather(receive_intel_task, report_sightings_task, match_task)
 
-    atexit.register(unsubscribe, zmq_endpoint, topic)
-    atexit.register(intel_task.cancel)
-    atexit.register(sighting_task.cancel)
-    await asyncio.gather(intel_task, sighting_task)
 
+def ioc_to_query(intel):
+    """
+    Creates a valid VAST query from an intel IoC item. See the Threat Bus
+    VAST plugin, `message_mapping.py`, for details of possible IoC `type`s.
+    """
+    ioc = intel.get("ioc", None)
+    type_ = intel.get("type", None)
+    if type_ == "ip" or type_ == "ipv6":
+        return f":addr == {ioc}"
+    if type_ == "url":
+        return f'"{ioc}" in url'
+    if type_ == "domain":
+        return f'"{ioc}" in domain || "{ioc}" in host || "{ioc}" in hostname'
```

Comment:
Along the same lines as the above comment: here it needs to be clear what the default schema you are targeting.

Reply:
Given that VAST does suffix matching, this line should cover most of the existing schemas for now. We can always extend that later / make it configurable. IMHO this is fine for now. What would you suggest?

Reply:
I think all this needs *right now* is a comment that explains what the basis for the conversion is, i.e., make it clear what the target schema is. (Zeek? Suricata? both?)

Reply:
Added a comment.